config/tectonic-node-controller/config.go

```diff
@@ -20,10 +20,12 @@ const (
 type ControllerConfig struct {
 	metav1.TypeMeta     `json:",inline"`
 	ClusterDNSIP        string `json:"clusterDNSIP"`
-	CloudProvider       string `json:"cloudProvider"`
 	CloudProviderConfig string `json:"cloudProviderConfig"`
 	ClusterName         string `json:"clusterName"`
 
+	// The "platform", e.g. "aws" or "libvirt"
+	InstallerPlatform string `json:"installerPlatform"`
```

Comment:
nit/bikeshed on the name. 

Provider?
Platform?
ProviderPlatform?

"Installer" is one aspect - but this value can be used through the lifecycle of the cluster. Even "InstallPlatform" might be slightly better. 

There is a good likelihood that this is a section of config users would interact with - so name should be enough that a user that is familiar with kubernetes that this would also enable the cloud-provider functionality (if a cloud-platform). with InstallerPlatform - I wouldn't guess that (but we can also put doc comments around the value).

Reply:
*shrug* Just duplicating what the other objects (utility, ingress) use. Diego chose it, you'll have to ask him :-)

Reply:
But Platform makes the most sense to me.